src/app/screens/Registration/index.tsx

```diff
@@ -1,89 +1,113 @@
 import React, { useCallback } from 'react';
 import i18next from 'i18next';
+import { useForm } from 'react-hook-form';
+import { useHistory } from 'react-router-dom';
 
-// import FormInput from '~components/FormInput';
+import { useDispatch } from '~contexts/UserContext';
+import { actionCreators, Credentials, User } from '~contexts/UserContext/reducer';
+import { login, setCurrentUser } from '~services/AuthServices';
+import { useLazyRequest } from '~app/hooks/useRequest';
+import FormInput from '~components/FormInput';
 import PATHS from '~components/Routes/paths';
-// import { stringArrayToObject } from '~utils/array';
+import { ERROR_MESSAGES } from '~constants/errorMessages';
 
-import styles from './styles.module.scss';
+import styles from '../Login/styles.module.scss';
 
-// const FIELDS = stringArrayToObject(['name', 'surname', 'email', 'password', 'confirmPassword']);
+import { VALIDATION_SCHEMA, FIELD_NAMES } from './constants';
 
-function RegistrationContainer() {
-  /* eslint-disable @typescript-eslint/no-unused-vars */
-  const handleSubmit = useCallback((event: React.FormEvent<Element>) => {
-    // TODO: Implement
-  }, []);
+function Registration() {
+  const { register, handleSubmit, errors, formState, getValues, setError, clearErrors } = useForm();
+  const { dirtyFields } = formState;
+  const history = useHistory();
+  const dispatch = useDispatch();
+  const [, , , loginRequest] = useLazyRequest({
+    request: (credentials: Credentials) => login(credentials),
+    withPostSuccess: response => {
+      // TODO integrate this when backend
+      const userResponse = response as User;
+      dispatch(actionCreators.setUser(userResponse));
+      setCurrentUser(userResponse);
+      history.push('/');
+    }
+  });
 
-  const handleInputChange = useCallback((event: React.FormEvent<Element>) => {
-    // TODO: Implement method
-  }, []);
+  const handleChangePasswordRepeat = useCallback(
+    e => {
+      const { password } = getValues();
+      const { value } = e.target;
+      if (value === password) {
+        return clearErrors('passwordRepeat');
+      }
+      setError(FIELD_NAMES.PASSWORD_REPEAT, { type: 'notMatch', message: ERROR_MESSAGES.passwordRepeat });
+      return 0;
+    },
+    [clearErrors, getValues, setError]
+  );
```

Comment:
Why do you need this callback? Can't you add a validation that depends on other fields?